### ISSUE TITLE
Fails to bring iTerm window to front

### ISSUE THREAD

**Issue by kolchurinvv** [vitorgalvao/custom-alfred-iterm-scripts#57]
Hey! I ran into a small issue where if my iTerm app isn't running and I issue any command, say `whoami` using `>` in alfred 4, the script launches the iTerm app, but the window app window never appears. It is there though. If I look at the window tab in the top panel or press `option+cmd+1` it'll pop up. Unfortunately the command isn't being run.

System: macOS 11.6
iTerm: Build 3.4.10
Alfred: 4.6 [1266]

**Comment by ShredderWalnut** [comment]
I believe I’m having the exact same issue. 

Symptoms: Using the script from the repo release, sometimes the command would run and sometimes it wouldn’t. After switching from the script in the repo release to the modified one [in this thread](https://github.com/vitorgalvao/custom-alfred-iterm-scripts/issues/57#issuecomment-945696952), the command would run with the iTerm window hidden.

macOS 12.0.1(21A559)
iTerm 3.4.14
Alfred 4.6.1 (1274)

**Comment by vitorgalvao** [comment]
Then you have the same weirdness in your setup. I cannot reproduce, so I cannot investigate further. You need to find the issue yourself.

**Comment by ShredderWalnut** [comment]
> I cannot reproduce, so I cannot investigate further.

Thank you for the reply! I *just* downloaded iTerm and I thought my setup should be quite vanilla. It’s a shame I can’t identify the underlying issue. If I *do* find something new, I’ll post an update here then.

**Comment by vitorgalvao** [comment]
> I _just_ downloaded iTerm and I thought my setup should be quite vanilla.

The problem is likely to be somewhere else in macOS (e.g. another tool interfering).

**Comment by ShredderWalnut** [comment]
That makes sense. That you for the tip!

**Comment by kolchurinvv** [comment]
@ShredderWalnut i found out what was causing my iTerm not to show the new window: _Settings > Profile > Window > Style._ My setting was set at "Full-height left of screen". As soon as I set it to "normal" the window popped up as expected

**Comment by ShredderWalnut** [comment]
Hi @kolchurinvv, thanks for the update. Curiously, my setting has always been “Normal”. I changed it to “Full-height left of screen”, relaunched it and changed it back to “normal”, and the issue still persisted.

I poked around in iTerm’s settings and found out if I change “Window restoration policy” (in Preferences > General > Startup) from “Use System Window Restoration Setting” to “Only Restore Hotkey Window”, the issue would be fixed for me. No idea what’s actually happening under the hood.
